fml/message_loop_task_queues.cc

```diff
@@ -140,9 +164,136 @@ void MessageLoopTaskQueues::Swap(TaskQueueId primary, TaskQueueId secondary)
 void MessageLoopTaskQueues::SetWakeable(TaskQueueId queue_id,
                                         fml::Wakeable* wakeable) {
   std::scoped_lock lock(GetMutex(queue_id, MutexType::kWakeables));
+  FML_CHECK(!wakeables_[queue_id]) << "Wakeable can only be set once.";
   wakeables_[queue_id] = wakeable;
 }
 
+bool MessageLoopTaskQueues::Merge(TaskQueueId owner, TaskQueueId subsumed) {
+  // task_observers locks
+  std::mutex& o1 = GetMutex(owner, MutexType::kObservers);
+  std::mutex& o2 = GetMutex(subsumed, MutexType::kObservers);
+
+  // delayed_tasks locks
+  std::mutex& t1 = GetMutex(owner, MutexType::kTasks);
+  std::mutex& t2 = GetMutex(subsumed, MutexType::kTasks);
+
+  std::scoped_lock(o1, o2, t1, t2);
```

Comment:
This does not actually perform the lock. Please use a variable here.

Reply:
Apparently I also missed the scoped_lock in the other review. Just global search and replace this pattern in the codebase.

Reply:
Done